webapp/components/NewsItem/NewsItem.vue

```diff
@@ -2,40 +2,48 @@
   <div>
     <div class="item-title">{{ item.title }} ({{ item.votes }})</div>
     <div class="item-buttons">
-      <button @click="upvote">Upvote</button>
-      <button @click="downvote">Downvote</button>
-      <button @click="remove">Remove</button>
+      <button v-if="loggedIn" @click="upvote">Upvote</button>
+      <button v-if="loggedIn" @click="downvote">Downvote</button>
+      <button v-if="isAuthor" @click="remove">Remove</button>
+      <button v-if="isAuthor" @click="edit">Edit</button>
```

Comment:
⭐Implemented Upvote and Downvote in logged in state, Remove and Edit only if it's the author. Optional: Only show the upvote or downvote button if you didn't upvoted / downvoted before.

Reply:
:heavy_plus_sign: :one: